src/megaapi_impl.cpp

```diff
@@ -2863,6 +2894,41 @@ char *MegaApiImpl::dumpXMPPSession()
     return buf;
 }
 
+char *MegaApiImpl::getAccountAuth()
+{
+    sdkMutex.lock();
+    char* buf = NULL;
+
+    if (client->loggedin())
+    {
+        buf = new char[MAX_SESSION_LENGTH * 4 / 3 + 4];
+        Base64::btoa((const byte *)client->sid.data(), client->sid.size(), buf);
+    }
+
+    sdkMutex.unlock();
+    return buf;
+}
+
+void MegaApiImpl::setAccountAuth(const char *auth)
+{
+    sdkMutex.lock();
+    if (!auth)
```

Comment:
We could protect this method against empty strings like in `MegaNodePrivate::setPrivateAuth()` by changing the condition by: `if (!auth || !auth[0])`